src/Hooks/useOAuth2Client.ts

```diff
@@ -0,0 +1,56 @@
+import {useCallback, useEffect, useState} from 'react';
+import {useGlobalLynxClient} from '@iotopen/react-lynx';
+import {ErrorResponse, OAuth2Client} from '@iotopen/node-lynx';
+
+export const useOAuth2Client = (id: string) => {
+    const {lynxClient} = useGlobalLynxClient();
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState<ErrorResponse | undefined>();
+    const [client, setClient] = useState<OAuth2Client>({
+        created: 0,
+        updated: 0,
+        name: '',
+        id: '',
+        client_secret: '',
+        trusted: false,
+        allowed_scopes: [],
+        icon_uri: '',
+        policy_uri: '',
+        tos_uri: '',
+        redirect_uris: [],
+    });
+    const refresh = useCallback(() => {
+        if(!loading) setLoading(true);
+        lynxClient.getOAuth2Client(id).then(client => {
+            if (error !== undefined) setError(undefined);
+            setClient(client);
+        }).catch(e => {
+            setError(e);
+        }).finally(() => {
+            setLoading(false);
+        });
+    }, [loading, lynxClient, id, error]);
+
+    useEffect(() => {
+        refresh();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [id]);
+
+    const remove = useCallback(() => {
+        return lynxClient.deleteOAuth2Client(client);
+    }, [client, lynxClient]);
+
+    const update = useCallback(() => {
+        return lynxClient.updateOAuth2Client(client);
+    }, [client, lynxClient]);
+
+    return {
+        refresh,
```

Comment:
Order here should be important since later ones are optional and can be skipped when used.

Se example here from useFunctions:
```
        loading: loading,
        error: error,
        create: create,
        remove: remove,
        functions: functions,
        refresh: refreshCall,
```
They are expected to be the same I would say

Reply:
order does not matter in objects.